clients/client-rust/src/credentials.rs

```diff
@@ -0,0 +1,376 @@
+                        err
+                    ))
+                }
+            },
+            Ok(cert) if cert.is_empty() => None,
+            Ok(cert) => Some(cert),
+        };
+
+        Ok(Credentials {
+            client_id,
+            access_token,
+            certificate,
+        })
+    }
+
+    /// Create a new Credentials object with clientId and accessToken
+    ///
+    /// Examples:
+    ///
+    /// ```
+    /// use taskcluster::Credentials;
+    /// let _ = Credentials::new("my_client_id", "my_access_token");
+    /// ```
+    pub fn new<S1: Into<String>, S2: Into<String>>(client_id: S1, access_token: S2) -> Credentials {
+        Credentials {
+            client_id: client_id.into(),
+            access_token: access_token.into(),
+            certificate: None,
+        }
+    }
+
+    /// Create a new Credentials object with clientId, accessToken, and certificate
+    ///
+    /// Examples:
+    ///
+    /// ```
+    /// use taskcluster::Credentials;
+    /// let _ = Credentials::new_with_certificate("my_client_id", "my_access_token", "{}");
+    /// ```
+    pub fn new_with_certificate<S1, S2, S3>(
+        client_id: S1,
+        access_token: S2,
+        certificate: S3,
+    ) -> Credentials
+    where
+        S1: Into<String>,
+        S2: Into<String>,
+        S3: Into<String>,
+    {
+        Credentials {
+            client_id: client_id.into(),
+            access_token: access_token.into(),
+            certificate: Some(certificate.into()),
+        }
+    }
+
+    /// Generate temporary credentials from permanent credentials, valid for the given duration,
+    /// starting immediately.  The temporary credentials' scopes must be a subset of the permanent
+    /// credentials' scopes. The duration may not be more than 31 days. Any authorized scopes of
+    /// the permanent credentials will be passed through as authorized scopes to the temporary
+    /// credentials, but will not be restricted via the certificate.
+    ///
+    /// Note that the auth service already applies a 5 minute clock skew to the
+    /// start and expiry times in
+    /// https://github.com/taskcluster/taskcluster-auth/pull/117 so no clock skew is
+    /// applied in this method, nor should be applied by the caller.
+    ///
+    /// See https://docs.taskcluster.net/docs/manual/design/apis/hawk/temporary-credentials
+    pub fn create_named_temp_creds(
+        &self,
+        temp_client_id: &str,
+        duration: Duration,
+        scopes: impl IntoIterator<Item = impl AsRef<str>>,
+    ) -> Result<Credentials, Error> {
+        self.create_temp_creds_inner(Some(temp_client_id), duration, scopes)
+    }
+
+    /// Similar to `create_named_temp_creds`, but creating unnamed credentials.  This approach is
+    /// still supported but users are encouraged to create named credentials when possible to
+    /// support auditability.
+    pub fn create_temp_creds(
```

Comment:
would `create_temp_creds` and `create_unnamed_temp_creds` better indicate the desired one? I'm happy either way